support/controlplane-component/defaults.go

```diff
@@ -86,6 +86,8 @@ func (c *controlPlaneWorkload[T]) setDefaultOptions(cpContext ControlPlaneContex
 	if err != nil {
 		return err
 	}
+	enforceTerminationMessagePolicy(podTemplateSpec.Spec.InitContainers)
+	enforceTerminationMessagePolicy(podTemplateSpec.Spec.Containers)
```

Comment:
To fix the e2e, we would need to move this call after we inject konnectivity/tokenMinter/avaialbiltiyProber containers so that those containers also get the new TerminationMessagePolicy. i.e. after 
https://github.com/openshift/hypershift/blob/e74a1bb4154e7b4e04e73d0e657711d25e1e4870/support/controlplane-component/defaults.go#L113-L119